File: 0819-most-common-word/0819-most-common-word.kt
class Solution {
    fun mostCommonWord(paragraph: String, banned: Array<String>): String {
        val counts: MutableMap<String, Int> = mutableMapOf()
        
        val words = paragraph.replace(Regex("\\W+"), " ").toLowerCase().split(" ").filter { it !in banned && it.isNotEmpty() }   
        
        for (word in words){
            counts[word] = counts.getOrDefault(word, 0) + 1
        }

        return counts.maxBy {it.value}!!.key
    }
}

File: 0819-most-common-word/0819-most-common-word.py
import re
from collections import Counter

class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        paragraph = filter(lambda x: (x not in banned), re.sub(r'[^\w]', ' ', paragraph).lower().split())
        return Counter(paragraph).most_common(1)[0][0]